### PR TITLE
Refactor events mechanics

### PR DESCRIPTION
**Pull request by radoslawlandowskikainos**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/AssignHearingPartToSessionAction.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import uk.gov.hmcts.reform.sandl.snlevents.service.RulesService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class AssignHearingPartToSessionAction extends Action implements RulesProcessable {
+
+    protected HearingPartSessionRelationship hearingPartSessionRelationship;
+    protected UUID hearingPartId;
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/AssignHearingPartToSessionAction.java]: private plox

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/AssignHearingPartToSessionAction.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        hearingPartRepository.save(hearingPart); // TODO: Extract into a separate 'persist' method?
+    }
+
+    @Override
+    public List<UserTransactionData> generateUserTransactionData() throws Exception {
+        List<UserTransactionData> userTransactionDataList = new ArrayList<>();
+        userTransactionDataList.add(new UserTransactionData("hearingPart",
+            hearingPart.getId(),
+            objectMapper.writeValueAsString(hearingPart),
+            "update",
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/AssignHearingPartToSessionAction.java]: Put it as a constant somewhere.

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/HearingPartController.java [hunk 3]
```diff
@@ -35,6 +39,15 @@
     @Autowired
     private RulesService rulesService;
 
+    @Autowired
+    HearingPartRepository hearingPartRepository;
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/HearingPartController.java]: private

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/HearingPartController.java [hunk 4]
```diff
@@ -70,10 +83,12 @@ public ResponseEntity upsertHearingPart(@RequestBody CreateHearingPart createHea
     @PutMapping(path = "/{hearingPartId}", consumes = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity assignHearingPartToSession(
         @PathVariable UUID hearingPartId,
-        @RequestBody HearingPartSessionRelationship assignment) throws IOException {
+        @RequestBody HearingPartSessionRelationship assignment) throws Exception {
 
-        UserTransaction ut = hearingPartService.assignHearingPartToSessionWithTransaction(hearingPartId, assignment);
+        UserTransaction ut = null; //hearingPartService.assignHearingPartToSessionWithTransaction(hearingPartId, assignment);
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/HearingPartController.java]: remove please

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/SessionController.java [hunk 5]
```diff
@@ -17,6 +17,7 @@
 import uk.gov.hmcts.reform.sandl.snlevents.model.request.UpsertSession;
 import uk.gov.hmcts.reform.sandl.snlevents.model.response.SessionInfo;
 import uk.gov.hmcts.reform.sandl.snlevents.model.response.SessionWithHearings;
+import uk.gov.hmcts.reform.sandl.snlevents.service.ActionService;
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/SessionController.java]: unused

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/ActionService.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class ActionService {
+
+    @Autowired
+    UserTransactionService userTransactionService;
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/ActionService.java]: private

**Comment by radoslawlandowskikainos** [comment]
The PR is closed as this branch was merged into antoher branch with #105

**Pull request closed without merging**